devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -0,0 +1,634 @@
+							options: it.values,
+							defaultValue: it.defaultValue,
+							required: false
+					)
+					break
+				case "range":
+					input(
+							name: it.key,
+							type: "number",
+							title: "Set value (range ${it.range})",
+							defaultValue: it.defaultValue,
+							range: it.range,
+							required: false
+					)
+					break
+			}
+		}
+		// Preferences template end
+	}
+}
+
+//Globals, input types used in sevice settings (parameter #1: Input 1 switch type)
+private getINPUT_TYPE_MONO_STABLE_SWITCH() {0}
+private getINPUT_TYPE_BI_STABLE_SWITCH() {1}
+private getINPUT_TYPE_POTENTIOMETER() {2}
+private getINPUT_TYPE_TEMPERATURE_SENSOR() {3}
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	// Preferences template begin
+	state.currentPreferencesState = [:]
+	parameterMap.each {
+		state.currentPreferencesState."$it.key" = [:]
+		state.currentPreferencesState."$it.key".value = getPreferenceValue(it)
+		if (it.type == "boolRange" && getPreferenceValue(it) == it.disableValue) {
+			state.currentPreferencesState."$it.key".status = "disablePending"
+		} else {
+			state.currentPreferencesState."$it.key".status = "synced"
+		}
+	}
+	readConfigurationFromTheDevice()
+	// Preferences template end
+}
+
+def updated() {
+// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	// Preferences template begin
+	parameterMap.each {
+		if (isPreferenceChanged(it)) {
+			log.debug "Preference ${it.key} has been updated from value: ${state.currentPreferencesState."$it.key".value} to ${settings."$it.key"}"
+			state.currentPreferencesState."$it.key".status = "syncPending"
+			if (it.type == "boolRange") {
+				def preferenceName = it.key + "Boolean"
+				if (notNullCheck(settings."$preferenceName")) {
+					if (!settings."$preferenceName") {
+						state.currentPreferencesState."$it.key".status = "disablePending"
+					} else if (state.currentPreferencesState."$it.key".status == "disabled") {
+						state.currentPreferencesState."$it.key".status = "syncPending"
+					}
+				} else {
+					state.currentPreferencesState."$it.key".status = "syncPending"
+				}
+			}
+		} else if (!state.currentPreferencesState."$it.key".value) {
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	syncConfiguration()
+	// Preferences template end
+}
+
+private readConfigurationFromTheDevice() {
+	def commands = []
+	parameterMap.each {
+		state.currentPreferencesState."$it.key".status = "reverseSyncPending"
+		commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
```

Comment:
you could just collect all of the commands and then call `encapCommands` on the whole list

Reply:
Fixed (I refactored the code, it should look better now).